prov/gni/src/gnix_eq.c

```diff
@@ -338,9 +333,16 @@ DIRECT_FN STATIC ssize_t gnix_eq_read(struct fid_eq *eq, uint32_t *event,
 
 	eq_priv = container_of(eq, struct gnix_fid_eq, eq_fid);
 
+	if ((blocking && !eq_priv->wait) ||
+	    (blocking && eq_priv->attr.wait_obj == FI_WAIT_SET))
```

Comment:
If FI_WAIT_SET is not supported, it should be prevented in gnix_eq_open(), not here.